async/asyncError.go

```diff
@@ -0,0 +1,41 @@
+package async
+
+// AsyncError is an async value that will eventually return an error
+// It is similair to a Promise/Future which returns an error
+// The value is supplied by calling SetValue.
+// Once the value is supplied AsyncError is considered completed
+// The value can be retrieved once AsyncError is completed via TryGetValue
+type AsyncError struct {
+	errCh chan error
+}
+
+func newAsyncError() *AsyncError {
+	return &AsyncError{
+		errCh: make(chan error, 1),
+	}
+}
+
+// Sets the value for the AsyncError.  Marks AsyncError as Completed or Fulfilled.
+// This method should only ever be called once per AsyncError instance.
+// Calling it more than once it will block indefinitely
+func (p *AsyncError) SetValue(err error) {
+	p.errCh <- err
+}
+
+// Returns the Status of this AsyncError, Completed or Pending
```

Comment:
eh I think it adds more complexity than its worth since there really are only two states.  Also TryGet() is a common api in C# on concurrent collections where a boolean specifies if it exists.  Also this is inline with golang where maps return bool, value